### PR TITLE
UnitTests and Coverage reporting

### PR DESCRIPTION
**Pull request by joenpera**
Jira-Id: SECO-4966

### CODE REVIEW

File: modules/sc-mesh-secure-deployment/src/nats/tests/test_settings.py
Comment (pentestiing) [inline modules/sc-mesh-secure-deployment/src/nats/tests/test_settings.py:1]: There are many tests here that check `self.assertEqual(ret, "FAIL", msg=f"ret: {ret}, mesh_status: {mesh_status}")` where `ret` and `mesh_status` come directly from the method that's being tested. While I think it's perfectly fine to use this to test the `FAIL` or `OK` status, I believe that in this case the content of the `mesh_status` message shall also be checked against a expected string :upside_down_face:

**Review by pentestiing — CHANGES_REQUESTED** [review]
Plz also check the `mesh_status` :page_with_curl:

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py:1]: 🏳️

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py [hunk 1]
```diff
@@ -65,21 +65,30 @@ def validate_mesh_settings(self, index: int) -> (str, str):
             return "FAIL", "Invalid mode"
         self.logger.debug("validate mesh settings mode ok")
 
-        if validation.validate_frequency(int(self.frequency[index])) is False:
+        try:
+            if validation.validate_frequency(int(self.frequency[index])) is False:
+                return "FAIL", "Invalid frequency"
+            self.logger.debug("validate mesh settings freq ok")
+        except ValueError:
```
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py:72]: I thought that it was prettier and easier to read.     I will fix all validations to check index and I remove the try/except.

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py [hunk 2]
```diff
@@ -65,21 +65,30 @@ def validate_mesh_settings(self, index: int) -> (str, str):
...
             return "FAIL", "Invalid frequency"
-        self.logger.debug("validate mesh settings freq ok")
 
-        if validation.validate_frequency(int(self.frequency_mcc[index])) is False:
+        try:
+            if validation.validate_frequency(int(self.frequency_mcc[index])) is False:
+                return "FAIL", "Invalid mcc frequency"
+            self.logger.debug("validate mesh settings mcc freq ok")
+        except ValueError:
```
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py:79]: same as 🔝

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py [hunk 3]
```diff
@@ -65,21 +65,30 @@ def validate_mesh_settings(self, index: int) -> (str, str):
...
         if validation.validate_country_code(self.country[index]) is False:
             return "FAIL", "Invalid country code"
         self.logger.debug("validate mesh settings country ok")
 
-        if validation.validate_tx_power(int(self.tx_power[index])) is False:
+        try:
+            if validation.validate_tx_power(int(self.tx_power[index])) is False:
+                return "FAIL", "Invalid tx power"
+            self.logger.debug("validate mesh settings tx power ok")
+        except ValueError:
```
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py:90]: same as 🔝🔝

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py:1]: [8c633fd](https://github.com/tiiuae/mesh_com/pull/444/commits/8c633fdc9a11f6873d5e258c3e7107a082615e72)

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py]: [8c633fd](https://github.com/tiiuae/mesh_com/pull/444/commits/8c633fdc9a11f6873d5e258c3e7107a082615e72)

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py]: [8c633fd](https://github.com/tiiuae/mesh_com/pull/444/commits/8c633fdc9a11f6873d5e258c3e7107a082615e72)

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py]: [8c633fd](https://github.com/tiiuae/mesh_com/pull/444/commits/8c633fdc9a11f6873d5e258c3e7107a082615e72)

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pentestiing) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py]: I won't say I'm against this usage of exceptions for accessing arrays by index (I use it myself), but I don't like the double `return <code>, <msg>` duplication as one might think there are differences between the `return` from inside the `if` and the one in the `except`, when in this case there aren't :waffle:

[the remark this answers is not in the document]
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py]: got your point.

File: modules/sc-mesh-secure-deployment/src/nats/tests/test_settings.py
Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/tests/test_settings.py:1]: [459af25](https://github.com/tiiuae/mesh_com/pull/444/commits/459af2525362c220e8dd675e7db24741c760d71f)

**Review by pentestiing — APPROVED** [review]
Great party! :tada:

**Review by saauvine — APPROVED** [review]

**Pull request merged**